clang/lib/Sema/SemaBounds.cpp

```diff
@@ -4197,12 +4204,18 @@ namespace {
       Expr *SubExpr = E->getSubExpr()->IgnoreParens();
       UnaryOperatorKind Op = E->getOpcode();
       
-      // Inverse(f, &*e1) = Inverse(f, e1)
       if (Op == UnaryOperatorKind::UO_AddrOf) {
+        // Inverse(f, &*e1) = Inverse(f, e1)
         if (UnaryOperator *UnarySubExpr = dyn_cast<UnaryOperator>(SubExpr)) {
           if (UnarySubExpr->getOpcode() == UnaryOperatorKind::UO_Deref)
             return Inverse(X, F, UnarySubExpr->getSubExpr());
         }
+        // Inverse(f, &e1[e2]) = Inverse(f, e1 + e2)
+        else if (ArraySubscriptExpr *ArraySubExpr = dyn_cast<ArraySubscriptExpr>(SubExpr))
+          return Inverse(X, F, ExprCreatorUtil::CreateBinaryOperator(S,
```

Comment:
This allocates a new binary operator expression in the AST context, which causes the expression to live forever.   We want to avoid analysis routines allocating indefinitely-lived data when possible.  You might want to allocate the binary operator in a local variable here (that is, on the stack) instead.